shared/naturalcrit/codeEditor/fold-code.js

```diff
@@ -0,0 +1,31 @@
+module.exports = {
+	registerHomebreweryHelper : function(CodeMirror) {
+		CodeMirror.registerHelper('fold', 'homebrewery', function(cm, start) {
+			const matcher = /^\\page.*/;
+			const prevLine = cm.getLine(start.line - 1);
+
+			if(start.line === cm.firstLine() || prevLine.match(matcher)) {
+				const lastLineNo = cm.lastLine();
+				let end = start.line, nextLine = cm.getLine(start.line + 1);
+
+				while (end < lastLineNo) {
+					if(nextLine.match(matcher)) {
+						return {
+							from : CodeMirror.Pos(start.line, 0),
+							to   : CodeMirror.Pos(end, cm.getLine(end).length)
+						};
+					}
+					++end;
+					nextLine = cm.getLine(end + 1);
+				}
+
+				return {
+					from : CodeMirror.Pos(start.line, 0),
+					to   : CodeMirror.Pos(end, cm.getLine(end).length)
+				};
```

Comment:
Just a simplification in the logic here (I *think* this still works)

```suggestion
				while (end < lastLineNo) {
					if(nextLine.match(matcher))
						break;

					++end;
					nextLine = cm.getLine(end + 1);
				}

				return {
					from : CodeMirror.Pos(start.line, 0),
					to   : CodeMirror.Pos(end, cm.getLine(end).length)
				};
```

Reply:
Last few little nitpicks. It does seem to work just fine, so this is just a matter of cleaning up a bit.

Reply:
Will be updated with my next push